.github/workflows/release.yaml

```diff
@@ -0,0 +1,67 @@
+name: release
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  goreleaser:
+    name: goreleaser
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Install Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: ~1.18
```

Comment:
FYI, since you're already using actions/checkout@v3 you can also have it determine your go version from your go.mod, for e.g.: https://github.com/operator-framework/operator-registry/blob/master/.github/workflows/build.yaml

Reply:
Oh dang, that's really nice!

I think we should create an issue for that and do it as a follow-up so we can update all of our actions to that at the same time.

Thanks for pointing that out @grokspawn !